### PR TITLE
Update schedule page and functionality to match new DB

### LINKED ISSUE #179: Scheduler: Update page and functionality to match new database
### Describe the feature

With the recent changes in our DB choice (changing some SQL to mongo), the website's scheduler page no longer works and needs some slight changes along with a small overhaul of a specific functionality within it, allowing the usage of mongo.

### Use Case

- When I use the schedule page to create a rule, I want to be capable of doing so without any problem so that I can feel confident that it works.

### Proposed Solution

The page's design has already been theorized and designed in figma, so all that there's left to do is apply it to the code.

### Other Information

_No response_

### PR DESCRIPTION
**Pull request by pdPinho**
### Issue

Closes #179 

### Reason for this change

Due to some changes in our database the backend required some adjustments whilst the frontend had to match those adjustments.

### Description of changes

- Updated `Schedule`, `ScheduleModal` and `ScheduleModalContent` to match new DB and backend.
- Updated and refactored some frontend code to use `react query`;
- Some miscellaneous fixes.

### Description of how you validated changes

Ran `docker compose up`
And on top of that tested multiple routes and functionalities to guarantee that things worked correctly.

### Checklist
- [X] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### Aditional Information
Currently missing some verifications but nothing software-breaking experience type.

### CODE REVIEW

File: frontend/src/routes/Schedule/priorityConfig.jsx
Comment (fungame2270) [inline frontend/src/routes/Schedule/priorityConfig.jsx:1]: isto ta pinta vou copiar isto po meu

File: frontend/src/components/Widgets/optionWidgetConfig.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const optionWidget = (templateWidget, setShowContentsPortal, setSelectedWidgetId, selectedContent) => {
+    /*
+            <div className="text-sm min-w-[90%] max-w-[90%]">
+
+        </div>
+    */
```
Comment (fungame2270) [inline frontend/src/components/Widgets/optionWidgetConfig.jsx]: unused code

**Review by fungame2270 — APPROVED** [review]

File: frontend/src/components/Widgets/optionWidgetConfig.jsx (hunk 1 above)
Comment (pdPinho) [reply frontend/src/components/Widgets/optionWidgetConfig.jsx]: Completely missed that. Thank you!

**Comment by pdPinho** [comment]
Code coverage complaint comes from DataLoader (which will be deleted in the future).
And some functions (Rule). These will be addressed in the near future.

**Pull request merged**